tests/dialects/test_hw.py

```diff
@@ -25,3 +46,132 @@ def test_inner_sym_target():
     assert sub_target
     assert not sub_target.is_op_only()
     assert sub_target.is_field()
+
+
+@irdl_op_definition
+class ModuleOp(IRDLOperation):
+    name = "module"
+    region: Region = region_def()
+    sym_name = attr_def(StringAttr)
+    traits = frozenset({InnerSymbolTable(), SymbolOpInterface()})
+
+
+@irdl_op_definition
+class OutputOp(IRDLOperation):
+    name = "output"
+    traits = frozenset({IsTerminator()})
+
+
+@irdl_op_definition
+class CircuitOp(IRDLOperation):
+    name = "circuit"
```

Comment:
Are these mimicking CIRCT operations? Is this from FIRRTL for example?
If so, do you mind adding a comment pointing that out?

Not sure if the plan is to actually implement them in the end in their respective dialects, but it might be helpful as placeholder comment.